index.js

```diff
@@ -2,58 +2,79 @@
      * @param  {Array|Object}   config.executor             Array of executors to load or a single executor object
+     * @param  {Object}         config.redisConnection      Connection details for redis
      * @param  {String}         config.executor[x].name     Name of the executor NPM module to load
      * @param  {String}         config.executor[x].options  Configuration to construct the module with
      */
     constructor(config = {}) {
-        const ecosystem = config.ecosystem || {};
         const executorConfig = config.executor;
-        const defaultPlugin = config.defaultPlugin;
+        const ecosystem = config.ecosystem || {};
 
         if (!executorConfig || !(Array.isArray(executorConfig)) || executorConfig.length === 0) {
             throw new Error('No executor config passed in.');
         }
 
         super();
 
-        executorConfig.forEach((plugin) => {
-            let ExecutorPlugin;
+        this.ecosystem = config.ecosystem || {};
+        this.defaultPlugin = config.defaultPlugin;
 
-            try {
-                // eslint-disable-next-line global-require, import/no-dynamic-require
-                ExecutorPlugin = require(`screwdriver-executor-${plugin.name}`);
-            } catch (err) {
-                console.error(err);
+        this.executorConfigs = {};
+        executorConfig.forEach((plugin) => {
+            this.executorConfigs[plugin.name] = Object.assign({ ecosystem }, plugin.options);
 
-                return;
+            if (!this.defaultPlugin) {
+                this.defaultPlugin = plugin.name;
             }
+        });
 
-            const options = Object.assign({ ecosystem }, plugin.options); // Add ecosystem to executor options
-            const executorPlugin = new ExecutorPlugin(options);
-
-            if (defaultPlugin && defaultPlugin === plugin.name) {
-                // set the default plugin to the desired one
-                this.defaultExecutor = executorPlugin;
-            } else if (!defaultPlugin && !this.defaultExecutor) {
-                // When given no default plugin, use the first module we instantiated
-                this.defaultExecutor = executorPlugin;
-            }
+        // eslint-disable-next-line new-cap
+        this.queue = new Resque.queue({ connection: config.redisConnection });
 
-            this[plugin.name] = executorPlugin;
-        });
+        // TODO: queue.connect is async, if we try to enqueue anything before it finishes it will fail
+        // Also, connect fails if not given a callback
+        this.queue.connect(() => {});
+    }
 
-        if (!this.defaultExecutor) {
-            throw new Error('No default executor set.');
-        }
+    /**
+     * Get the full config for a job
+     * @method _getFullConfig
+     * @param {Object} buildConfig  build configuration
+     * @return {Object}
+     */
+    getFullConfig(buildConfig) {
```

Comment:
The JSDoc says the method name is `_getFullConfig` and this function is defined as `getFullConfig` (one does not have an underscore). Which one is the correct name?